src/utils/TestUser.ts

```diff
@@ -0,0 +1,50 @@
+import path from 'node:path';
+
+import _ from 'lodash';
+
+import RoleType from '@/constants/RoleTypes';
+import AppConfig from '@/utils/AppConfig';
+
+class TestUser {
+  username: string;
+  password: string;
+  storagePath: string;
+  requiredRoles: Set<RoleType>;
+
+  constructor(
+    username: string,
+    password: string,
+    storageFileName: string,
+    requiredRoles: Set<RoleType>
+  ) {
+    this.username = username;
+    this.password = password;
+    this.storagePath = path.join(
+      AppConfig.AUTH_STORAGE_DIR_PATH,
+      storageFileName
+    );
+    this.requiredRoles = requiredRoles;
+  }
+
+  assertAllRequiredRoles(userRoles: Set<string>) {
+    const unexpectedRoles = _.difference([...userRoles], [...this.requiredRoles]);
+    const absentRequiredRoles = _.difference([...this.requiredRoles], [...userRoles]);
```

Comment:
Are those spread operators necessary?

Reply:
`userRole` and `requiredRoles` are of type `Set` and `_.difference()` takes in arrays